test/functional/helpers.lua

```diff
@@ -291,15 +291,51 @@ local function write_file(name, text, dont_dedent)
   file:close()
 end
 
-local function source(code)
-  local tmpname = os.tmpname()
-  if os_name() == 'osx' and string.match(tmpname, '^/tmp') then
-   tmpname = '/private'..tmpname
+-- Tries to get platform name, from $CMAKE_SYSTEM_NAME, uname,
+-- fallback is 'Windows'
+local uname = (function()
+  local platform = nil
+  return (function()
+    if platform then
+      return platform
+    end
+
+    platform = os.getenv("CMAKE_SYSTEM_NAME")
```

Comment:
It's not that important, but maybe we could name this just `SYSTEM_NAME` or `NVIM_SYSTEM_NAME` or something, it's not really related to CMake.